route/pool_test.go

```diff
@@ -231,6 +232,110 @@ var _ = Describe("EndpointPool", func() {
 
 		})
 	})
+	Context("Load Balancing Algorithm of a pool", func() {
+
+		It("has a value specified in the pool options", func() {
+			poolWithLBAlgo := route.NewPool(&route.PoolOpts{
+				Logger:                 logger,
+				LoadBalancingAlgorithm: config.LOAD_BALANCE_RR,
+			})
+			Expect(poolWithLBAlgo.LoadBalancingAlgorithm).To(Equal(config.LOAD_BALANCE_RR))
+		})
+
+		It("has an invalid value specified in the pool options", func() {
+			poolWithLBAlgo2 := route.NewPool(&route.PoolOpts{
+				Logger:                 logger,
+				LoadBalancingAlgorithm: "wrong-lb-algo",
+			})
+			iterator := poolWithLBAlgo2.Endpoints(logger, "", false, "none", "zone")
+			Expect(iterator).To(BeAssignableToTypeOf(&route.RoundRobin{}))
+			Expect(logger.Buffer()).To(gbytes.Say(`invalid-pool-load-balancing-algorithm`))
+		})
+
+		It("is correctly propagated to the newly created endpoints LOAD_BALANCE_LC ", func() {
+			poolWithLBAlgoLC := route.NewPool(&route.PoolOpts{
+				Logger:                 logger,
+				LoadBalancingAlgorithm: config.LOAD_BALANCE_LC,
+			})
+			iterator := poolWithLBAlgoLC.Endpoints(logger, "", false, "none", "az")
+			Expect(iterator).To(BeAssignableToTypeOf(&route.LeastConnection{}))
+			Expect(logger.Buffer()).To(gbytes.Say(`endpoint-with-least-connection-lb-algo-added-to-pool`))
+		})
+
+		It("is correctly propagated to the newly created endpoints LOAD_BALANCE_RR ", func() {
+			poolWithLBAlgoLC := route.NewPool(&route.PoolOpts{
+				Logger:                 logger,
+				LoadBalancingAlgorithm: config.LOAD_BALANCE_RR,
+			})
+			iterator := poolWithLBAlgoLC.Endpoints(logger, "", false, "none", "az")
+			Expect(iterator).To(BeAssignableToTypeOf(&route.RoundRobin{}))
+			Expect(logger.Buffer()).To(gbytes.Say(`endpoint-with-round-robin-lb-algo-added-to-pool`))
+		})
+	})
+
+	Context("Load balancing algorithm of a newly added endpoint", func() {
+
+		It("is valid and will overwrite the load balancing algorithm of a pool", func() {
+			pool := route.NewPool(&route.PoolOpts{
+				Logger:                 logger,
+				LoadBalancingAlgorithm: config.LOAD_BALANCE_RR,
+			})
+			expectedLBAlgo := config.LOAD_BALANCE_LC
+			endpoint := route.NewEndpoint(&route.EndpointOpts{
+				Host: "host-1", Port: 1234,
+				RouteServiceUrl:        "url",
+				LoadBalancingAlgorithm: expectedLBAlgo,
+			})
+			pool.OverrulePoolLoadBalancingAlgorithm(endpoint)
+			Expect(pool.LoadBalancingAlgorithm).To(Equal(expectedLBAlgo))
+			Expect(logger.Buffer()).To(gbytes.Say(`setting-pool-load-balancing-algorithm-to-that-of-an-endpoint`))
+		})
+
+		It("is an empty string and the load balancing algorithm of a pool is kept", func() {
+			expectedLBAlgo := config.LOAD_BALANCE_RR
+			pool := route.NewPool(&route.PoolOpts{
+				Logger:                 logger,
+				LoadBalancingAlgorithm: expectedLBAlgo,
+			})
+			endpoint := route.NewEndpoint(&route.EndpointOpts{
+				Host: "host-1", Port: 1234,
+				RouteServiceUrl:        "url",
+				LoadBalancingAlgorithm: "",
```

Comment:
Same as above, this test and the one below are equivalent.

Reply:
done